### DISCUSSION TITLE
ESP32 3.0.0-a appears to  break TFT_eSPI on at least ESP32C3 with GC9A01 driver

### DISCUSSION THREAD

**Discussion by Shaolo** [Bodmer/TFT_eSPI#3129] [Alerts]
I  can't  speak  to other combos, but I made the mistake of starting with 3.0.0-a instead of 2.0.14.

Really long story short, on 3.0.0-a , it will compile,  won't run w/o USE_HSPI_PORT, but will only draw a black screen. If USE_HSPI_PORT  isn't declared, then the  code hangs in the init function.

Roll back to 2.0.14 if you're seeing this and banging your  head against the wall.

This  is also a working config for the ESP32C3 boards with the little round display attached on AliExpress that  use the GC9A01 driver.

> // See SetupX_Template.h for all options available
> #define USER_SETUP_ID 200
> 
> #define GC9A01_DRIVER
> 
> #define TFT_WIDTH 240 // GC9A01 240 x 240
> #define TFT_HEIGHT 240 // GC9A01 240 x 240
> 
> #define TFT_SCLK 6
> #define TFT_MOSI 7 // In some display driver board, it might be written as "SDA" and so on.
> #define TFT_DC   2  // Data Command control pin
> 
> #define TFT_CS   10  // Chip select control pin
> 
> #define TFT_RST  -1  // Reset pin (could connect to Arduino RESET pin)
> 
> #define TFT_BL   3  // LED back-light
> #define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)
> 
> #define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
> #define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
> #define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
> #define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
> #define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
> #define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
> #define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
> 
> #define SMOOTH_FONT
> 
> #define SPI_FREQUENCY  80000000
> 
> // Optional reduced SPI frequency for reading TFT
> #define SPI_READ_FREQUENCY  20000000

![SS-3065 (6)](https://github.com/Bodmer/TFT_eSPI/assets/949654/6d4bfe16-fed7-4ca1-b45e-531dc2767e55)

As an aside, LovyanGFX was working with 3.0.0-a and that's part of what took me so long to figure this out. Hope this helps someone else.

**Comment by Bodmer** [comment]
When 3.0.0 has been debugged and released I will see if any changes to the library are required.

**Comment by Basitadam** [comment]
I had the same issue wtih 3.0.2 as well. Runs OK with 3.0.1, 3.0.3 and 3.0.4

**Comment by vishalsoniindia** [comment]
the issue is solve by defining 

#define USE_HSPI_PORT          // Force HSPI for esp32 v3.0 or above

https://github.com/vishalsoniindia/TFT_ESPI_not_working_with_esp32_v3.0.0_Above
